src/aiida/storage/sqlite_dos/backend.py

```diff
@@ -313,6 +313,16 @@ def nodes(self):
     def users(self):
         return orm.SqliteUserCollection(self)
 
+    def set_global_variable(
+        self, key: str, value: None | str | int | float, description: str | None = None, overwrite=True
+    ) -> None:
+        from sqlalchemy.exc import OperationalError
+
+        try:
+            super().set_global_variable(key, value, description, overwrite)
+        except OperationalError:
```

Comment:
Ideally, I would try to make sure the error here is really due to a locked database, but I am not sure that is possible in a deterministic way apart from parsing the error message, but that feels too fragile.